gcp_airflow_foundations/common/utils/task.py

```diff
@@ -0,0 +1,8 @@
+from airflow.models import TaskInstance
+
+
+def is_first_task_execution(task_id, dag_id, ):
+    ti = TaskInstance(task_id=task_id,
+                      dag_id=dag_id,
+                      execution_date=execution_date)
+    return ti.previous_ti() is None
```

Comment:
Is this related to the use case you mentioned last week about detecting whether a DAG had its first execution completed?